sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryClientBuilder.java

```diff
@@ -0,0 +1,295 @@
+        return this;
+    }
+
+    /**
+     * Sets The configuration store that is used during construction of the service client.
+     *
+     * @param configuration the configuration value.
+     * @return the ContainerRepositoryClientBuilder.
+     */
+    public ContainerRepositoryClientBuilder configuration(Configuration configuration) {
+        this.configuration = configuration;
+        return this;
+    }
+
+    /**
+     * Sets The logging configuration for HTTP requests and responses.
+     *
+     * @param httpLogOptions the httpLogOptions value.
+     * @return the ContainerRepositoryClientBuilder.
+     */
+    public ContainerRepositoryClientBuilder httpLogOptions(HttpLogOptions httpLogOptions) {
+        this.httpLogOptions = Objects.requireNonNull(httpLogOptions, "'logOptions' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Sets The retry policy that will attempt to retry failed requests, if applicable.
+     *
+     * @param retryPolicy the retryPolicy value.
+     * @return the ContainerRepositoryClientBuilder.
+     */
+    public ContainerRepositoryClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        this.retryPolicy = retryPolicy;
+        return this;
+    }
+
+    /**
+     * Adds a custom Http httpPipeline policy.
+     *
+     * @param customPolicy The custom Http httpPipeline policy to add.
+     * @return the ContainerRepositoryClientBuilder.
+     */
+    public ContainerRepositoryClientBuilder addPolicy(HttpPipelinePolicy customPolicy) {
+        Objects.requireNonNull(customPolicy, "'pipelinePolicy' cannot be null.");
+
+        if (customPolicy.getPipelinePosition() == HttpPipelinePosition.PER_CALL) {
+            perCallPolicies.add(customPolicy);
+        } else {
+            perRetryPolicies.add(customPolicy);
+        }
+
+        return this;
+    }
+
+    /**
+     * Builds an instance of ContainerRegistryImpl with the provided parameters.
+     *
+     * @return an instance of ContainerRegistryImpl.
+     */
+    public ContainerRepositoryAsyncClient buildAsyncClient() {
+        if (serializerAdapter == null) {
+            this.serializerAdapter = JacksonAdapter.createDefaultSerializerAdapter();
+        }
+
+        // Service version
+        ContainerRegistryServiceVersion serviceVersion = (version != null)
+            ? version
+            : ContainerRegistryServiceVersion.getLatest();
+
+        if (httpPipeline == null) {
+            this.httpPipeline = createHttpPipeline();
+        }
+
+        ContainerRepositoryAsyncClient client = new ContainerRepositoryAsyncClient(repository, httpPipeline, serializerAdapter, endpoint, serviceVersion.getVersion());
+        return client;
+    }
+
+    private HttpPipeline createHttpPipeline() {
+        Configuration buildConfiguration =
+            (configuration == null) ? Configuration.getGlobalConfiguration() : configuration;
+        if (httpLogOptions == null) {
+            httpLogOptions = new HttpLogOptions();
```

Comment:
Does the service have a well-known set of safe HTTP headers and query parameters that can be logged? If it does a default logging policy allow listing those should be created, view the builders in Blob Storage or Search for more details.